google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/hive/HiveCredentialTester.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.tsunami.plugins.detectors.credentials.genericweakcredentialdetector.testers.hive;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.tsunami.common.net.http.HttpRequest.get;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.common.net.HostAndPort;
+import com.google.tsunami.common.data.NetworkEndpointUtils;
+import com.google.tsunami.common.net.db.ConnectionProviderInterface;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpStatus;
+import com.google.tsunami.plugins.detectors.credentials.genericweakcredentialdetector.provider.TestCredential;
+import com.google.tsunami.plugins.detectors.credentials.genericweakcredentialdetector.tester.CredentialTester;
+import com.google.tsunami.proto.NetworkService;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/** Credential tester specifically for hive. */
+public final class HiveCredentialTester extends CredentialTester {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private final ConnectionProviderInterface connectionProvider;
+  private final HttpClient httpClient;
+  private static final String HIVE_TITLE = "<title>HiveServer2</title>";
+
+  @Inject
+  HiveCredentialTester(ConnectionProviderInterface connectionProvider, HttpClient httpClient) {
+    this.connectionProvider = checkNotNull(connectionProvider);
+    this.httpClient = httpClient;
+  }
+
+  @Override
+  public String name() {
+    return "HiveCredentialTester";
+  }
+
+  @Override
+  public String description() {
+    return "Hive credential tester.";
+  }
+
+  @Override
+  public boolean canAccept(NetworkService networkService) {
+    HostAndPort targetPage = NetworkEndpointUtils.toHostAndPort(networkService.getNetworkEndpoint());
+    String targetUri = String.format("http://%s:%d", targetPage.getHost(), 10002);
```

Comment:
I'm not 100% sure of this part. By running the plugin i'm noticing that it tries to connect with jdbc hive to every open port that the nmap scan finds. Can you please investigate into it and fix it?

Reply:
You should gain information from the fingerprint phase, in order to understand if Apache Hive is running. When the results says so (for example because you have found the `<title>HiveServer2</title>` string) you can start bruteforcing the credentials. 

The flow should be the following:
- First you fingerprint with HTTP
- You can bruteforce the credentials (using jdbc) over the result of the fingerprint phase

Reply:
I don't think that is necessary to try connecting with jdbc to every open ports, it will only generate unnecessary noise. You should stick with the default port.

Reply:
In my opinion, you should fingerprint with HTTP the results of the nmap scan. Then if there is a success you can use jdbc on the port 10000. How does it sounds?